yt_dlp/extractor/screencastify.py

```diff
@@ -0,0 +1,49 @@
+import urllib.parse
+
+from .common import InfoExtractor
+from ..utils import traverse_obj, update_url_query
+
+
+class ScreencastifyIE(InfoExtractor):
+    _VALID_URL = r'https?://watch\.screencastify\.com/v/(?P<id>[^/?#]+)'
+    _TESTS = [{
+        'url': 'https://watch.screencastify.com/v/sYVkZip3quLKhHw4Ybk8',
+        'info_dict': {
+            'id': 'sYVkZip3quLKhHw4Ybk8',
+            'ext': 'mp4',
+            'title': 'Inserting and Aligning the Case Top and Bottom',
+            'description': '',
+            'uploader': 'Paul Gunn',
+            'extra_param_to_segment_url': str,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        info = self._download_json(
+            f'https://umbrella.svc.screencastify.com/api/umbrellaService/watch/{video_id}', video_id)
+
+        query_string = traverse_obj(info, ('manifest', 'auth', 'query'))
+        query = urllib.parse.parse_qs(query_string)
+        formats = []
+        hls_manifest_url = traverse_obj(info, ('manifest', 'hlsUrl'))
+        if hls_manifest_url:
+            formats.extend(
+                self._extract_m3u8_formats(
+                    hls_manifest_url, video_id, ext='mp4', m3u8_id='hls', query=query))
+        # DASH formats also need query to be appended to segment URLs but downloader is unable to
+        # dash_manifest_url = traverse_obj(info, ('manifest', 'url'))
```

Comment:
Supporting extra_param_to_segment_url in dash.py sounds trivial. Am I missing something?

Reply:
you're right. should I make a new PR for that or commit it to this one?

Reply:
Same PR is fine, but I suggest adding it to #5528